xdsl/dialects/onnx.py

```diff
@@ -0,0 +1,105 @@
+from __future__ import annotations
+
+from typing import Annotated, TypeVar
+
+from xdsl.dialects.builtin import (
+    SSAValue,
+    TensorType,
+)
+from xdsl.ir import (
+    Attribute,
+    Dialect,
+)
+from xdsl.irdl import (
+    ConstraintVar,
+    IRDLOperation,
+    irdl_op_definition,
+    operand_def,
+    result_def,
+)
+from xdsl.utils.exceptions import VerifyException
+
+_T = TypeVar("_T", bound=Attribute)
+
+
+class ElementwiseBinOpBase(IRDLOperation):
+    """Base class for element-wise binary operations on tensors with Numpy-style broadcasting."""
+
+    T = Annotated[Attribute, ConstraintVar("T"), _T]
```

Comment:
```suggestion
    T = Annotated[Attribute, ConstraintVar("T")]
```
Since you don't use `_T` in `ElementwiseBinOpBase`, you don't really need this.

Reply:
Related to this, I tried to figure out how to specify that the element type should be any of a specific subset of builtin types -- is there a good way to do this?

Reply:
Yes, just replace the `Attribute` with the type you want, something like:
```
T = Annotated[Float32Type | IntegerType, ConstraintVar("T")]
```

Reply:
Thanks! Changed to `T = Annotated[AnyFloat | IntegerType, ConstraintVar("T")]`, which matches the spec more closely.